core/arch/arm/mm/core_mmu.c

```diff
@@ -959,6 +959,10 @@ static void check_pa_matches_va(void *va, paddr_t pa)
 		return;
 	}
 #ifdef CFG_WITH_PAGER
+	if (v >= CFG_TEE_LOAD_ADDR && v < core_mmu_linear_map_end) {
+		TEE_ASSERT(v == pa);
```

Comment:
`TEE_ASSERT()` is unconditional, it will always evaluate and check the expression.
`assert()` is supposed to depend on `NDEBUG` and only evaluate and check expressions if `NDEBUG` isn't defined.

Our current `assert()` implementation doesn't honor `NDEBUG`, but may be changed into that some day.

I'm on the other hand tempted to remove `TEE_ASSERT()` and replace it with explicit `if (!expr) panic()`, that's something we would need to discuss first of course.